packages/design/src/DataTableNext/types.ts

```diff
@@ -17,6 +18,9 @@ type TableColumnBase<T> = {
 export type PaginationConfig = {
   pageSize?: number;
   pagerPosition?: 'top' | 'bottom';
+};
+
+export type FetchingConfig = {
```

Comment:
I think it's better if `onFetchMore` and `fetchStatus` are both part of TableProps without a dedicated (which also optional) `FetchingConfig` type. But I have no strong preferences.

Reply:
I think grouping them up makes it more clear that they come together. Similar to how we do for pagination. Also, this way we can make the fetching config optional, but require both fields in case the user does want to add it.